### PR TITLE
Add EventsIterator support for Prophesee camera

### PR DESCRIPTION
**Pull request by hexu33**

### CODE REVIEW

File: tests/lava/lib/peripherals/dvs/test_prophesee.py [hunk 1]
```diff
@@ -25,15 +25,23 @@
...
-    PyPropheseeCameraModel,
-)
 from lava.lib.peripherals.dvs.transformation import Compose, Downsample
 from metavision_core.utils import get_sample
-from metavision_core.event_io import RawReader, EventDatReader
+from metavision_core.event_io import (
+    RawReader, EventDatReader, EventsIterator)
 from metavision_sdk_cv import ActivityNoiseFilterAlgorithm
 
+USE_EVENTSITERATOR = True
```
Comment (mathisrichter) [inline tests/lava/lib/peripherals/dvs/test_prophesee.py:34]: There need to be unit tests for both the ProcessModel using the RawReader as well as the one using the EventsIterator. A global switch for the unit tests that would have to be changed manually before running the tests does not work here.

Please make sure that there are separate tests for the different ProcessModel implementations and that the correct ProcessModels are imported for those tests only.

**Review by mathisrichter — CHANGES_REQUESTED** [review]
This needs some higher-level refactoring. We will do a finer-grained review of the code and unit tests once my points have been addressed.

File: src/lava/lib/peripherals/dvs/prophesee_rawreader.py [hunk 2]
```diff
@@ -24,6 +25,7 @@
 from lava.lib.peripherals.dvs.transformation import Compose, EventVolume
 
 from metavision_core.event_io import RawReader, EventDatReader
+from metavision_core.event_io import EventsIterator
```
[the remark this answers is not in the document]
Comment (seowwj) [reply src/lava/lib/peripherals/dvs/prophesee_rawreader.py]: Removed unused imports.

File: src/lava/lib/peripherals/dvs/prophesee_rawreader.py [hunk 3]
```diff
@@ -12,6 +12,7 @@
 
 import numpy as np
 import time
+from threading import Thread
```
[the remark this answers is not in the document]
Comment (seowwj) [reply src/lava/lib/peripherals/dvs/prophesee_rawreader.py]: Removed unused imports.

File: src/lava/lib/peripherals/dvs/prophesee_eventsiterator.py [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+        )
+
+
+class EventsIteratorWrapper():
+    """
+    PropheseeEventsIterator class for PropheseeCamera which will create a
+    thread in the background to always grab events within a time window and
+    put them in a buffer.
+    
+    delta_t: Control the size of the time window for collecting events,
```
[the remark this answers is not in the document]
Comment (seowwj) [reply src/lava/lib/peripherals/dvs/prophesee_eventsiterator.py]: We will merge in `EventsIterator` and `RawReader` into a single file, with tagging.

The caveat with tagging is that the current implementation of `select_tag` in `run_config` would require that all other ProcessModels (and not just the Prophesee ProcessModel) be tagged appropriately. For instance, if they use the tag `event_it` then the same tag needs to appear in other ProcessModels as well.

Currently the best way to do select the ProcessModel is to manually create a custom process model map and pass it to the run configuration.

File: tests/lava/lib/peripherals/dvs/test_prophesee.py [hunk 5]
```diff
@@ -50,6 +58,18 @@
...
+    if USE_EVENTSITERATOR:
+        mv_iterator = EventsIterator(input_path=file_name,
+                                     delta_t=1000)
+        height, width = mv_iterator.get_size()
+        del mv_iterator
+    else:
+        reader = RawReader(file_name)
+        height, width = reader.get_size()
+        del reader
+    return height,width
```
[the remark this answers is not in the document]
Comment (seowwj) [reply tests/lava/lib/peripherals/dvs/test_prophesee.py]: Fixed

File: src/lava/lib/peripherals/dvs/prophesee_rawreader.py [hunk 6]
```diff
@@ -0,0 +1,239 @@
+# Copyright (C) 2023 Intel Corporation
```
Comment (mathisrichter) [inline src/lava/lib/peripherals/dvs/prophesee_rawreader.py:1]: Is this file still needed? You seem to have ProcessModels for both the EventsIterator and RawReader implementation in the other file.

File: tests/lava/lib/peripherals/dvs/test_prophesee.py (hunk 1 above)
Comment (mathisrichter) [reply tests/lava/lib/peripherals/dvs/test_prophesee.py:34]: This comment has not been addressed, yet. As it is, only the ProcessModel with the EventsIterator implementation is being tested by the unit tests. Please remove this global flag and write dedicated tests for the PropheseeCamera Process with its different ProcessModels (i.e., RawReader and EventsIterator).

If there is something blocking you, please do not hesitate to reach out and discuss that with me.

**Review by mathisrichter — CHANGES_REQUESTED** [review]

**Comment by PhilippPlank** [comment]
Hello, I have started a new PR for this change (#18) to enable our automatic unit test runs. Currently our CI pipeline does not work for PRs coming from public forks... 

Please use #18 for future changes! Thanks.

**Pull request closed without merging**